### PR TITLE
Update rectangle extension

### LINKED ISSUE #13075: [Rectangle] ...
### Extension

https://www.raycast.com/crickford/rectangle

### Description

This is fantastic, but I would love support for Rectangle Pro :) 

### Who will benefit from this feature?

Users of Rectangle Pro.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by crickford**
## Description

Adds support for Rectangle Pro (closes #13075)

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/rectangle/src/actions/rectangle-pro.test.ts [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+  for (const action of rectangleActions) {
+    if (!proActions.has(action)) {
+      detectedMissingActions.add(action);
+    }
+  }
+
+  expect(detectedMissingActions).toEqual(knownMissingActions);
+});
+
+// test("All large icons used", () => {
```
Comment (mil3na) [inline extensions/rectangle/src/actions/rectangle-pro.test.ts:96]: If not needed, please remove this test

**Review by mil3na — CHANGES_REQUESTED** [review]
Hi!
Thank you for your contribution! My name is Milena and I will be reviewing it today.

I do not have access to Rectangle Pro, so I could not test this, but have you made sure that all new commands work? Are they marked differently on the UI?

**Comment by mil3na** [comment]
Also, thank you so much for adding tests! It made my day :)

File: extensions/rectangle/src/actions/rectangle-pro.test.ts (hunk 1 above)
Comment (crickford) [reply extensions/rectangle/src/actions/rectangle-pro.test.ts:96]: Ah, good catch. I'll remove it

**Comment by crickford** [comment]
> Also, thank you so much for adding tests! It made my day :)

You bet! I appreciate the thoughtful review :)

**Comment by crickford** [comment]
> Hi! Thank you for your contribution! My name is Milena and I will be reviewing it today.

Hi Milena!

> I do not have access to Rectangle Pro, so I could not test this, but have you made sure that all new commands work? Are they marked differently on the UI?

I have tested them out w/ Rectangle Pro. Actually, there are no new commands - Rectangle Pro implements a superset of Rectangle's functionality, so the existing commands work for either Rectangle / Rectangle Pro. 

For the new _actions_ available via the `Rectangle Actions` command when using Rectangle Pro, the only visual differentiator is the search text placeholder. Perhaps I can add a screenshot showing the Rectangle Pro actions, with a "Rectangle Pro support included" banner - what do you think? 

It does seem that 2 of the _actions_ are not currently working, due to a [known issue](https://github.com/rxhanson/RectanglePro-Community/issues/463) on the Rectangle Pro side: 
1. `specified`
2. `fullscreen`

For number 1, the corresponding command is disabled by default. I have a todo item to document how to set it up (requires changes to Rectangle's settings) in the backlog for a future update :)
For number 2, I can remove that action for Rectangle Pro until an upstream fix is available.

**Comment by mil3na** [comment]
Hey! A screenshot would be great.
Also, if you know something will already not work, it is better to remove it to avoid confusion and angry users.

**Comment by crickford** [comment]
Hi @mil3na! I made the requested changes, removed the broken action and added a screenshot showing Rectangle Pro support. I do have some further ideas improvements I'd like to add in the future, but since it might be a bit before I can get to them I'd like to release the changes in this PR first. Should be all set to merge - mind taking another look? :)

**Review by mil3na — APPROVED** [review]
LGTM!

**Pull request merged**